pandas/core/generic.py

```diff
@@ -4932,6 +4932,7 @@ def filter(
         like: str | None = None,
         regex: str | None = None,
         axis=None,
+        multi="all"
```

Comment:
xref #23541. would prefer to see a `level` kwarg added.

Reply:
Hey @simonjayhawkins I changed the argument to level, while keeping original code incase we need to extend functionality in the future. I think its better we address the enhancement #23541 with this PR, it would solve the bug by extension or at least have similar functionality.